### ISSUE TITLE
Version control breaks with out-of-source builds

### ISSUE THREAD

**Issue by MBulli** [MBulli/SmartCommandlineArgs#14]
When using a tool like CMake the typical folder structure of a project differs from the default Visual Studio way. Usually only the cmake files are version controlled and the msbuild project files are generated from these files. Thus the project files aren't and shouldn't be under version control. Normally a folder like 'build' contains all generated project files and is wholly ignored by a vcs:

```
project/
|--  build/ [Ignored by vcs]
|    |-- project.sln
|    |-- project.csproj
|    +-- project.args.json
|-- src/
+-- CMakeLists.txt
```

Now our commandline arguments are stored inside the json files which are located in the same folder as the project files therefore are ignored by vcs, just like the project. This renders our vcs support useless for out of source builds.

**Comment by Irame** [comment]
A workaround would be  to create symbolic links from the source to the build directory with cmake:
```cmake
function(link_cmd_args_json_current_project project_name)
	if (UNIX)
		set (link "${CMAKE_CURRENT_BINARY_DIR}/${project_name}.args.json")
		set (target "${CMAKE_CURRENT_SOURCE_DIR}/${project_name}.args.json")
		set(command ln -s ${target} ${link})
	else()
		STRING(REGEX REPLACE "/" "\\\\" CMAKE_CURRENT_BINARY_DIR_WIN ${CMAKE_CURRENT_BINARY_DIR})
		STRING(REGEX REPLACE "/" "\\\\" CMAKE_CURRENT_SOURCE_DIR_WIN ${CMAKE_CURRENT_SOURCE_DIR})
		set (link "${CMAKE_CURRENT_BINARY_DIR_WIN}\\${project_name}.args.json")
		set (target "${CMAKE_CURRENT_SOURCE_DIR_WIN}\\${project_name}.args.json")
		set(command cmd.exe /c mklink ${link} ${target})
	endif()
	
	execute_process(COMMAND ${command} 
					RESULT_VARIABLE result
                    ERROR_VARIABLE output)
	
	if (NOT ${result} EQUAL 0)
		message(FATAL_ERROR "Could not create symbolic link for: ${target} --> ${output}")
    endif()
endfunction(link_cmd_args_json_current_project)
```
This method should be called where 'add_executable()' is called and the executable/project name should be passed:
```cmake
add_executable(MyProject ${CPP_FILES} ${H_FILES})
link_cmd_args_json_current_project(MyProject)
```

The only downside to this "solution" is that changes in the Json file are not tracked and therefore the extension is not updated upon changes in the file but the extension should update the file just fine.

**Comment by MBulli** [comment]
Fixed in v1.2.2 ( #cb1d5b2bd7ac7bb215cb689566165586c43731dd) and described in wiki section.
